core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -131,6 +132,35 @@ class KryoSerializer(conf: SparkConf)
     // our code override the generic serializers in Chill for things like Seq
     new AllScalaRegistrar().apply(kryo)
 
+    // Register types missed by Chill.
+    kryo.register(classOf[Array[Tuple1[Any]]])
```

Comment:
@rxin  yes, there is a check for (Int, Int) in the KryoSuite and without registering Array[Tuple2[Any, Any]] that test fails.